server/service/src/sync/translations/name.rs

```diff
@@ -209,6 +227,109 @@ impl SyncTranslation for NameTranslation {
 
         Ok(result)
     }
+
+    fn try_translate_push_upsert(
+        &self,
+        connection: &StorageConnection,
+        changelog: &ChangelogRow,
+    ) -> Result<Option<Vec<RemoteSyncRecordV5>>, anyhow::Error> {
+        if !match_push_table(changelog) {
+            return Ok(None);
+        }
+
+        let NameRow {
+            id,
+            name,
+            code,
+            r#type,
+            is_customer,
+            is_supplier,
+            supplying_store_id,
+            first_name,
+            last_name,
+            gender,
+            date_of_birth,
+            phone,
+            charge_code,
+            comment,
+            country,
+            address1,
+            address2,
+            email,
+            website,
+            is_manufacturer,
+            is_donor,
+            on_hold,
+            created_datetime,
+            is_deceased,
+            national_health_number,
+        } = NameRowRepository::new(connection)
+            .find_one_by_id(&changelog.record_id)?
+            .ok_or(anyhow::Error::msg(format!(
+                "Name row ({}) not found",
+                changelog.record_id
+            )))?;
+
+        let legacy_row = LegacyNameRow {
+            id,
+            name,
+            code,
+            r#type: from_name_type(&r#type),
```

Comment:
Hmm, we are only allowed to edit `Patient` records on the remote server, I think we should check this after finding the record in database (return error if not `Patient` type). And we wouldn't need `from_name_type` at this point

Reply:
Should I remove the from_name_type or just comment it out (just in case we do push other name types in the future)?

Reply:
I would remove it, we haven't had the need to push any other `name` types from remote servers in mSupply.

Sorry I think i was wrong with `return error`, should just be `None` ?